configure.ac

```diff
@@ -1021,7 +1052,9 @@ M0_CPPFLAGS_DIST=$(echo "$M0_CPPFLAGS" | grep -Po -e '-D\S+|-I/usr\S+' | xargs e
 # __attribute__, such as __attribute__((gccxml(...))). It's important because we
 # use -Werror, which turns warnings into errors, but we still need to be able to
 # use gccxml attributes.
-M0_CFLAGS="-pipe -Wall -Werror -Wno-attributes -fno-strict-aliasing $M0_CFLAGS"
+PAGE_SIZE=$(getconf PAGE_SIZE)
+PAGE_SHIFT=$(( $((PAGE_SIZE==4096)) ? 12 :  $(($((PAGE_SIZE==65536)) ? 16 :  14)) ))
+M0_CFLAGS="-pipe -Wall -Werror -Wno-attributes -fno-strict-aliasing -DPAGE_SIZE=$PAGE_SIZE -DPAGE_SHIFT=$PAGE_SHIFT $M0_CFLAGS"
```

Comment:
Why is this done through command line -D and not through config.h?

Reply:
I wonder - can we get the page size via `sysconf(_SC_PAGE_SIZE)` call in C-code? (See https://stackoverflow.com/a/3351960/3812944.)

Reply:
anyway this platform size checking is  redundant, if any page is not defined then code in configure.ac platform chaing will take and subsequent  macro  will take care
 "#ifndef PAGE_SHIFT
  #ifdef  CONFIG_X86_64
    #define PAGE_SHIFT    12
  #elif defined  CONFIG_AARCH64
    #define PAGE_SHIFT    16 
    /* Page Size on aarch64 is 64KB, hence Page Shift will be 16 */
  #else
    #error  "The platform is not supported"
  #endif
  #define PAGE_SIZE	  (1UL << PAGE_SHIFT)
#endif "

Reply:
If this is redundant - why to keep it?

Reply:
Shall i delete it from configure.ac file ?

Reply:
If it builds fine without it - yes, of course!

Reply:
No it is giving error

Reply:
What error?

Reply:
PAGE_SIZE undefined error

Reply:
When compiling which file? Could you copy/paste the exact error here, please?

Reply:
@nikitadanilov, fixed it via config.h.

Reply:
Oops, now it fails to compile:

```
In file included from ./arch/arm64/include/asm/memory.h:28:0,
                 from ./arch/arm64/include/asm/thread_info.h:30,
                 from ./include/linux/thread_info.h:38,
                 from ./arch/arm64/include/asm/preempt.h:5,
                 from ./include/linux/preempt.h:78,
                 from ./include/linux/spinlock.h:51,
                 from ./include/linux/seqlock.h:36,
                 from ./include/linux/time.h:6,
                 from ./include/linux/stat.h:19,
                 from ./include/linux/module.h:10,
                 from /root/motr/motr/linux_kernel/m0tr_main.c:24:
./arch/arm64/include/asm/page-def.h:26:0: error: "PAGE_SHIFT" redefined [-Werror]
 #define PAGE_SHIFT  CONFIG_ARM64_PAGE_SHIFT

In file included from <command-line>:0:0:
/root/motr/config.h:314:0: note: this is the location of the previous definition
 #define PAGE_SHIFT 16

In file included from ./arch/arm64/include/asm/memory.h:28:0,
                 from ./arch/arm64/include/asm/thread_info.h:30,
                 from ./include/linux/thread_info.h:38,
                 from ./arch/arm64/include/asm/preempt.h:5,
                 from ./include/linux/preempt.h:78,
                 from ./include/linux/spinlock.h:51,
                 from ./include/linux/seqlock.h:36,
                 from ./include/linux/time.h:6,
                 from ./include/linux/stat.h:19,
                 from ./include/linux/module.h:10,
                 from /root/motr/motr/linux_kernel/m0tr_main.c:24:
./arch/arm64/include/asm/page-def.h:28:0: error: "PAGE_SIZE" redefined [-Werror]
 #define PAGE_SIZE  (_AC(1, UL) << PAGE_SHIFT)

In file included from <command-line>:0:0:
/root/motr/config.h:317:0: note: this is the location of the previous definition
 #define PAGE_SIZE 65536
```

Reply:
Reverted the change back with a comment.

Reply:
Fixed by removing as not needed after #1109.